cirq/circuits/circuit_operation_test.py

```diff
@@ -480,6 +545,33 @@ def test_decompose_loops():
     assert cirq.Circuit(cirq.decompose_once(op)) == expected_circuit
 
 
+def test_decompose_loops_with_measurements():
+    a, b = cirq.LineQubit.range(2)
+    circuit = cirq.FrozenCircuit(
+        cirq.H(a),
+        cirq.CX(a, b),
+        cirq.measure(a, b, key='m'),
+    )
+    base_op = cirq.CircuitOperation(circuit)
+
+    op = base_op.with_qubits(b, a).repeat(3)
+    expected_circuit = cirq.Circuit(
+        cirq.H(b),
+        cirq.CX(b, a),
+        cirq.measure(b, a, key='0-m'),
+        cirq.H(b),
+        cirq.CX(b, a),
+        cirq.measure(b, a, key='1-m'),
+        cirq.H(b),
+        cirq.CX(b, a),
+        cirq.measure(b, a, key='2-m'),
+    )
+    assert cirq.Circuit(cirq.decompose_once(op)) == expected_circuit
+
+    with pytest.raises(ValueError, match='circuit is not invertible'):
+        _ = base_op.repeat(-2)
```

Comment:
Is this required here? There's another test that covers this, IIRC.

Reply:
Right. Was trying to contrast with the non-measurement test above but it's not really needed.